tests/models/nn/transformers/test_utils.py

```diff
@@ -0,0 +1,75 @@
+#  Copyright 2022-2025 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# pylint: disable=attribute-defined-outside-init
+
+import typing as tp
+
+import pandas as pd
+import pytest
+
+from rectools import Columns
+from rectools.dataset import Interactions
+from rectools.models.nn.transformers.utils import leave_one_out_mask
+
+
+class TestUtils:
+    @pytest.fixture
+    def interactions(self) -> Interactions:
+        df = pd.DataFrame(
+            [
+                [1, 1, 1, "2021-09-01"],  # 0
+                [1, 2, 1, "2021-09-02"],  # 1
+                [1, 1, 1, "2021-09-03"],  # 2
+                [1, 2, 1, "2021-09-04"],  # 3
+                [1, 3, 1, "2021-09-05"],  # 4
+                [2, 3, 1, "2021-09-06"],  # 5
+                [2, 2, 1, "2021-08-20"],  # 6
+                [2, 2, 1, "2021-09-06"],  # 7
+                [3, 1, 1, "2021-09-05"],  # 8
+                [1, 6, 1, "2021-09-05"],  # 9
+            ],
+            columns=[Columns.User, Columns.Item, Columns.Weight, Columns.Datetime],
+        ).astype({Columns.Datetime: "datetime64[ns]"})
+        return Interactions(df)
+
+    @pytest.mark.parametrize(
+        "swap_interactions,expected_val_index, expected_val_item, val_users",
+        (
+            ([9, 9], [7, 8, 9], 6, None),
+            ([9, 9], [7, 8, 9], 6, 3),
```

Comment:
Didn't test slicing for n users below the total number of users. Let's write `2` as val_users for this case